spec/Section 5 -- Validation.md

```diff
@@ -565,6 +569,41 @@ fragment conflictingDifferingResponses on Pet {
 }
 ```
 
+The same is true if a field is designated `Non-Nullable` in an operation. In this
+case, `someValue` could be either a `String` or a `String!` which are two different
+types and therefor can not be merged:
+
+```graphql counter-example
+fragment conflictingDifferingResponses on Pet {
+  ... on Dog {
+    someValue: nickname
+  }
+  ... on Cat {
+    someValue: nickname!
+  }
+}
+```
+
+### Client Controlled Nullability Designator List Dimensions
+
+**Formal Specification**
+
+* For each {field} in the document
+  * Let {fieldDef} be the definition of {field}
+  * Let {fieldType} be the type of {fieldDef}
+  * Let {requiredStatus} be the required status of {field}
+  * Let {designatorDepth} be the number of square bracket pairs in {requiredStatus}
+  * Let {typeDepth} be the number of list dimensions in {fieldType}
```

Comment:
Do I need to be more specific than this? I noticed `ASTVisitor` isn't part of the spec, so should I leave the implementation of this up to the reader's discretion, or should I describe how `ASTVisitor` functions in greater detail?

Reply:
You could write a ListDepth(type) algorithm that recursively unwraps the type and adds one if it was a list type; however I think describing this process inline would probably be fine - something like “let listCount be the number of list wrapper types encountered whilst recursively unwrapping type”

Reply:
(Written on mobile; do reference similar text to craft this more accurately)